src/Hooks.ts

```diff
@@ -16,9 +16,21 @@ export function useNavigation<S>(): NavigationScreenProp<S & NavigationRoute> {
 }
 
 export function useNavigationParam<T extends keyof NavigationParams>(
-  paramName: T
+  paramName: T,
+  fallbackValue?: NavigationParams[T]
 ) {
-  return useNavigation().getParam(paramName);
+  const { getParam, setParams } = useNavigation();
+  const setParamValue = useCallback(
+    () => {
+      return (newValue: NavigationParams[T]) =>
+        setParams({ [paramName]: newValue });
+    },
+    [setParams]
+  );
+  return [
+    fallbackValue ? getParam(paramName, fallbackValue) : getParam(paramName),
+    setParamValue(),
+  ];
 }
```

Comment:
```
export default function useNavigationParam<
  T extends keyof Params,
  Params extends NavigationParams = NavigationParams
>(
  paramName: T,
  fallback?: Params[T],
): [Params[T], (value: Params[T]) => boolean] {
  const { getParam, setParams } = useNavigation();
  const setParamValue = useCallback(() => {
    return (newValue: Params[T]) => setParams({ [paramName]: newValue });
  }, [paramName, setParams]);

  // had to cast `paramName` because `navigation.getParam` thinks that some other type can be passed
  return [getParam(paramName as string, fallback), setParamValue()];
}
```
I would suggest these change where I added additional optional type `Params` for `NavigationParams`, so that you can optionally determine actual types of keys' values instead of seeing `any` (it is made possible with `Index types` http://www.typescriptlang.org/docs/handbook/advanced-types.html#index-types). Also added `paramName` to deps as `ESlint` suggested